### ISSUE TITLE
[authoring] cursor rendered in wrong location

### ISSUE THREAD

**Issue by wittjosiah** [dxos/dxos#7466]
https://github.com/user-attachments/assets/221d84c2-8ff0-4670-9b11-880ca7340caf

**Comment by wittjosiah** [comment]
Potentially related to #6075

**Comment by thure** [comment]
This is likely related to #7465, since cursors are DOM nodes that intercede between spans of text, breaking soft wrapping.

**Comment by wittjosiah** [comment]
@thure it's possible, but this one I actually spent a bunch of time on this morning and I think it might be an issue with our automerge integration. I'm leaving this one unblocked because I want to look further into that with Dima soon.

**Comment by wittjosiah** [comment]
I've observed the wrong index being computed for where the cursor should be rendered so it's actually be placed _after_ the newline when it should be _before_ the newline. It may be a race condition between the cursor and replication and codemirror view or something along those lines.

**Comment by thure** [comment]
@wittjosiah Fair enough, it wouldn’t be related to wrapping if the cursor is in the wrong place entirely in the DOM.
